pkg/util/hardware/hw_utils.go

```diff
@@ -178,3 +179,6 @@ func LookupDeviceVCPUAffinity(pciAddress string, domainSpec *api.DomainSpec) ([]
 	}
 	return alignedVCPUList, nil
 }
+func TruncateSCSIDiskSerial(serial string) string {
+	return string([]byte(serial)[:MaxSCSISerialLen])
+}
```

Comment:
I think that the function should avoid runtime errors, even if the len is already checked by the caller
```suggestion
func TruncateSCSIDiskSerial(serial string) string {
	if len(serial) <= MaxSCSISerialLen{
		return serial
	}
	return string([]byte(serial)[:MaxSCSISerialLen])
}
```